backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+        """
+        self._create_rated_entity(field1=self.field, language="fr")
+        self._create_rated_entity(field1=self.field, language="fr")
+
+        response = self.client.get(self.url, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], self.setup_results_count)
+
+    def test_multiple_languages(self):
+        """
+        If the recommendation filter includes multiple languages,
+        each entity should have a relevance score equal to the
+        max relevance score over all languages.
+        """
+        self._create_rated_entity(field1=self.field, language="en")
+        self._create_rated_entity(field1=self.field, language="fr")
+
+        response = self.client.get(
+            self.url_base + f"?metadata[language]=en&metadata[language]=fr&search={self.query}",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 2 + self.setup_results_count)
+
+    def test_languages_unknown_by_postgres(self):
+        """
+        Postgres full-text search does not support as many languages
+        as the Tournesol query languages filter.
+
+        If all the languages of the query are not supported by Postgres,
+        a classic case-insensitive, language-independent search is done.
+
+        Just check that the search still works.
+        """
+        word = "kook"
+        self._create_rated_entity(field1=word, language="et")
+
+        response = self.client.get(
+            self._make_url(word, "et"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_sorting_depends_on_total_score(self):
+        """
+        If the relevance is the same, the results with higher
+        total score (or "weighted score") should appear first.
+        """
+        scores = [.3, -4.7, 6.0, 0.0, 8.1, -10.0, -2.4, -2.3, 1.2, 1.1]
+        entities = []
+        for score in scores:
+            entities.append(self._create_rated_entity(field1=self.field, criteria_score=score))
+
+        entities = [self.setup_entity] + entities
+        scores = [self.setup_score] + scores
+        self.assertEqual(len(scores), len(entities))
+
+        response = self.client.get(self.url + "&unsafe=true", format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], len(scores))
+        self.assertEqual(len(response.data["results"]), response.data["count"])
+
+        sorted_scores = sorted(scores, reverse=True)
+        for index, score in enumerate(sorted_scores):
+            entity = entities[scores.index(score)]
+            self.assertEqual(response.data["results"][index]["uid"], entity.uid)
+
+    def test_public_recommendations_sorting_depends_on_total_score(self):

```

Comment:
I'd add the term "personal" here.

`def test_public_personal_reco_sorting_depends_on_total_score`

Reply:
I'll use the term "contributor" instead of "personal", because that's what is used elsewhere

Reply:
Even if "personal" is arguably more adapted